### PR TITLE
Support downstream building

### PR DESCRIPTION
**Pull request by fiasco**
This is a pull request for #101 to allow downstream projects to make customisations within a sites directory in the root of the repository. govCMS doesn't not track this sites folder so it will need additional documenting.

### CODE REVIEW

**Comment by fiasco** [comment]
change built ok, tests passed, I'd be keen to get a final once over from @typhonius  or @srowlands before committing.

@teamglenny / @aleayr from a governance standpoint, this is an enabler for downstream users of govCMS, such as though who may not be on the SaaS product to contribute back to govCMS as a part of their downstream project. It also assists with making [this documentation](https://github.com/govCMS/govCMS/wiki/Tracking-govCMS-customisations-on-site-builds) easier to follow

**Comment by fiasco** [comment]
would like to get @teamglenny's perspective from a product standpoint and @srowlands technical perspective.

**Comment by gollyg** [comment]
Needs a rebase.

File: build/phing/build.xml [hunk 1]
```diff
@@ -238,6 +242,28 @@
...
+        <then>
+          <copy todir="${govcms.sites}">
+            <fileset dir="${docroot.sites}">
+                <include name="**"/>
+            </fileset>
+          </copy>
+        </then>
+      </if>
+
+      <!-- Must remote both directory (1st time attempt) and symlink (every other attempt) -->
```
Comment (gollyg) [inline build/phing/build.xml]: Typo - s/remote/remove

File: build/phing/build.xml [hunk 2]
```diff
@@ -238,6 +242,28 @@
...
+        <phingcall target="make:sites" />
+    </target>
+
+    <target name="make:sites">
+      <!-- Setup the sites directory outside of docroot so that future compilations
+           don't remove customisations make to the sites directory -->
+      <property name="sites.available" value="false" />
+      <available file="${govcms.sites}" property="sites.available" type="dir" />
+      <if>
+        <isfalse value="${sites.available}"/>
```
Comment (gollyg) [inline build/phing/build.xml]: Style for closing slash - not sure if we have a style guide for this? There is a mix of space before the closing slash and no space throughout this file. Would be good to standardise.

**Pull request closed without merging**